src/mock-doc/node.ts

```diff
@@ -297,7 +304,7 @@ export class MockElement extends MockNode {
 
   override cloneNode(_deep?: boolean): MockElement {
     // implemented on MockElement.prototype from within element.ts
-    return null;
+    return this;
```

Comment:
this is a little tricky
- we may signal to users that we're "cloning" a node when we're really not here (after all, it returned a value, why wouldn't they think it's a clone?)
- this may be considered a breaking change since we're changing the return value here

this may be one of the few cases where we want to return `null` (although IDK quite what the ramifications are there)

Reply:
yeah I wasn't entirely sure how to proceed here, however I'm not sure that in normal use this method will ever get called - it's monkey-patched here:

https://github.com/ionic-team/stencil/blob/e6c8078b1d8c13505bc9691eb2d11b7fc1df7964/src/mock-doc/element.ts#L596-L616

which now makes me wonder why it's even here on `MockElement` 🤔🤔🤔

Reply:
Yeah, I'm not sure either. Beyond that comment on `element.ts#cloneNode` I don't have the full story myself.

Maybe the best thing to do here (which I really don't love) is to keep it as `null` and use a type assertion? 😨 

```suggestion
    // @ts-ignore - implemented on MockElement.prototype from within element.ts
    return null as unknown as MockElement;
```

Reply:
sometimes

![](https://media3.giphy.com/media/PyLZ7uLdf96pi/giphy.gif?cid=ecf05e47wcpuofb8wturwn31dm9rqpq5rmx93n9wlojh5obr&rid=giphy.gif&ct=g)

Reply:
Never before has Futurama been so apt 😆 😢